src/components/progress-flow/progress-flow-item/progress-flow-item.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    h,
+    Prop,
+} from '@stencil/core';
+import { FlowItem } from '../progress-flow.types';
+
+/**
+ * @private
+ */
+@Component({
+    tag: 'limel-progress-flow-item',
+    shadow: false,
+    styleUrl: 'progress-flow-item.scss',
+})
+export class ProgressFlowItem {
+    @Element()
+    public element: HTMLLimelProgressFlowItemElement;
+
+    /**
+     * What flow item that should be rendered
+     */
+    @Prop()
+    public item: FlowItem = null;
+
+    /**
+     * True if the flow item should be disabled
+     */
+    @Prop()
+    public disabled = false;
+
+    /**
+     * True if the flow item should be readonly
+     */
+    @Prop()
+    public readonly = false;
+
+    /**
+     * Fired when clicking on the flow item
+     */
+    @Event()
+    public interact: EventEmitter<void>;
+
+    /**
+     * Icon displayed along with the text optionally
+     */
+    @Prop()
+    public icon: string;
+
+    constructor() {
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    public render() {
+        let secondaryText = this.item.secondaryText;
+        if (secondaryText) {
+            secondaryText = ' · ' + secondaryText;
+        }
+
+        const tooltip = this.item.text + secondaryText;
```

Comment:
The `secondaryText` is added to the tooltip, because both are used as a `title` attribute, which is displayed on `hover`. The reason that both are there is because the steps have a `max-width` to prevent them from growing too large. Also make better layouts on smaller containers. However, if text is too long, we can at least display it on hover, to enable users read it fully. If the `secondaryText` is optional, would it still say undefined? 

How do we solve this? Can you please suggest a code change?